### PR TITLE
Add Err msg when Sentry RLPx port is occupied #1398

### PR DESCRIPTION
**Pull request by claudioutt**
Also improved occupied port 9090 error message.

### CODE REVIEW

File: silkworm/infra/grpc/server/server.hpp [hunk 1]
```diff
@@ -82,7 +82,7 @@ class Server {
         server_ = builder.BuildAndStart();
         SILK_TRACE << "Server " << this << " bound at selected port: " << selected_port;
         if (server_ == nullptr) {
-            SILK_ERROR << "Server " << this << " BuildAndStart failed [" << settings_.address_uri << "]";
+            SILK_ERROR << "Server " << this << " BuildAndStart failed. Cannot start gRPC server at [" << settings_.address_uri << "] Port is probably in use!";
```
Comment (battlmonstr) [inline silkworm/infra/grpc/server/server.hpp:85]: is there a way to get an error from the builder?

File: silkworm/sentry/rlpx/server.cpp [hunk 2]
```diff
@@ -59,7 +59,11 @@ Task<void> Server::run(
     acceptor.set_option(detail::socket_option::boolean<SOL_SOCKET, SO_REUSEPORT>(true));
 #endif
 
-    acceptor.bind(endpoint);
+    boost::system::error_code ec;
+    acceptor.bind(endpoint, ec);
+    if(ec){
+        SILK_ERROR << "Sentry RLPx server has failed to start. TCP port: "<< port_ << " is busy. Try another port with --port=MMM.";
```
Comment (battlmonstr) [inline silkworm/sentry/rlpx/server.cpp:65]: * we use exceptions, please catch this as `const boost::system::system_error&`
* the check should examine the ex.code() and/or message and understand that it is related to the busy port
* throw `std::runtime_error` in the end, no need to log here

**Review by battlmonstr — CHANGES_REQUESTED** [review]

**Comment by battlmonstr** [comment]
please run `make fmt` and commit for the CI to pass

File: silkworm/infra/grpc/server/server.hpp (hunk 1 above)
Comment (claudioutt) [reply silkworm/infra/grpc/server/server.hpp:85]: I looked into exactly that. builder.BuildAndStart() just returns nullptr when an Error occurs.

Comment (battlmonstr) [reply silkworm/infra/grpc/server/server.hpp:85]: Try to implement [GlobalCallbacks::AddPort](https://github.com/grpc/grpc/blob/0b1e381d56da805045ce4c89d47ca4f13549c251/include/grpcpp/server.h#L86). Call `grpc::Server::SetGlobalCallbacks(...)` to register your implementation. `BuildAndStart()` calls it [here](https://github.com/grpc/grpc/blob/0b1e381d56da805045ce4c89d47ca4f13549c251/src/cpp/server/server_builder.cc#L434) and that should trigger a [callback](https://github.com/grpc/grpc/blob/master/src/cpp/server/server_cc.cc#L1131). If `port == 0` in the `AddPort` callback, we can assume that the port is busy.

Also please enhance the `std::runtime_error` thrown below with this extra information.

**Review by battlmonstr — CHANGES_REQUESTED** [review]

**Comment by battlmonstr** [comment]
@claudioutt do you have time to finish this in the upcoming weeks? We're considering to merge a part that is ready (rlpx), and keep the unfinished part (grpc) as a task for newcomers.

**Comment by battlmonstr** [comment]
superceded by https://github.com/erigontech/silkworm/pull/1788

**Comment by claudioutt** [comment]
Sorry I am currently really busy with work. Feel free to open the issue for new contributors.

**Pull request closed without merging**